### DISCUSSION TITLE
SEB 3.3.2 Installation Error related to WebView2 Bootstrapper

### DISCUSSION THREAD

**Discussion by Tanmaya-Code-Library** [SafeExamBrowser/seb-win-refactoring#424] [Q&A]
I am getting installation Error, while installing latest version of SEB (SEB 3.3.2). I am using the installer SEB_3.3.2.413_SetupBundle.exe.

**Setup Failed screen of SEB:-**
![image](https://user-images.githubusercontent.com/77824094/172030120-f1197bcf-c710-46b2-a9b4-19f14be7edea.png)

**Error logged in the installer log file:-** 
[3DF4:3784][2022-06-03T15:18:33]i301: Applying execute package: WebView2_Bootstrapper, action: Install, path: C:\ProgramData\Package Cache\366288994801930C7748750811DB9E9CE2B5295B\WebView2 Bootstrapper, arguments: '"C:\ProgramData\Package Cache\366288994801930C7748750811DB9E9CE2B5295B\WebView2 Bootstrapper" /silent /install'
[3DF4:3784][2022-06-03T15:18:36]e000: Error 0x8004090d: Process returned error: 0x8004090d
[3DF4:3784][2022-06-03T15:18:36]e000: Error 0x8004090d: Failed to execute EXE package.
[36D4:A7BC][2022-06-03T15:18:36]e000: Error 0x8004090d: Failed to configure per-machine EXE package.
[36D4:A7BC][2022-06-03T15:18:36]i319: Applied execute package: WebView2_Bootstrapper, result: 0x8004090d, restart: None
[36D4:A7BC][2022-06-03T15:18:36]e000: Error 0x8004090d: Failed to execute EXE package.
[3DF4:3784][2022-06-03T15:18:36]i351: Removing cached package: WebView2_Bootstrapper, from path: C:\ProgramData\Package Cache\366288994801930C7748750811DB9E9CE2B5295B\

I have started noticing this WebView2 Bootstrapper issue from 3rd June 2022.This issue was not there till 2nd June 2022.
In SEB 3.3.1 release, one fix was given for WebView2 installation issue. But we are still getting the WebView2 installation issue in the latest version of SEB installer. Kindly provide a permanent solution for this issue.

![image](https://user-images.githubusercontent.com/77824094/172030392-f92deca4-02e6-4aec-baf0-b3a941ea5655.png)

**Comment by dbuechel** [comment]
Could you please post the entire log file of the installer? As far as we know, this is not a general issue with the installer (tested it just now, works fine with installing the WebView2 dependency).

**Comment by Enriquerojlo** [comment]
Hi

I'm installing safe exam browser on the computers of my college and I find the same issue, but not in my personal computer.

Here is all the log file

[Safe_Exam_Browser_20220621134917.log](https://github.com/SafeExamBrowser/seb-win-refactoring/files/8948787/Safe_Exam_Browser_20220621134917.log)

The issue is when it tries to install WebView2_Bootstrapper 

Do you know how to solve it?

BR

**Reply by dbuechel** [reply]
This appears to be a system-specific issue. Searching for a solution to error code 0x80072efe yields various results, you might want to try those who appear appropriate. Otherwise, you could try to install SEB manually, i.e. first install all the required dependencies as listed in the README here on GitHub and then install the MSI setup of SEB for your respective platform (x86 or x64).

**Comment by NickETH** [comment]
Hi folks,
at least the error from @tanmaya-hub seems clear:
Error 0x8004090d means, "there is already a newer version of WV2 installed"
@dbuechel: WV2 installers are... (not easy going). You have to check for things like this, if you bundle it with SEB.
@Enriquerojlo: Could it be, that you tried to install the package to the user, instead of the machine?

Good luck,
Nick

**Reply by dbuechel** [reply]
Excellent, thanks for the hint @NickETH. I've created an issue (#429) and will improve that for the next release version. Do you happen to have a list with all possible return codes?

**Comment by NickETH** [comment]
This should help: https://www.thewindowsclub.com/fix-microsoft-edge-installation-update-errors
And because everything around MS Edge Chromium is literally Google stuff, 
this too: https://support.google.com/chrome/answer/6315198?hl=en
For me, testing seems to be essential.
Always check the log: msedge_installer.log (in Windows\temp if you install with system or just %temp% when installing from a logged on user)

**Reply by dbuechel** [reply]
Great, thanks, will try to handle these in the setup bundle.

**Comment by wirapratamaz** [comment]
Hey folks, I’m running into an issue installing the latest Safe Exam Browser in 2024, and it looks like it’s related to the WebView2 Bootstrapper.

Here’s a quick fix that worked for me:

1. **Install Microsoft Edge WebView2 Manually:**
   - Grab the offline installer from this link:  
     [Microsoft Edge WebView2](https://developer.microsoft.com/id-id/microsoft-edge/webview2?form=MA13LH)
   - Make sure to pick the right version (x86 or x64) for your device.

2. **Run as Admin:**
   - Right-click the installer and select "Run as administrator." Let it finish installing.

3. **Check Installation:**
   - Head to **Control Panel** > **Programs and Features** and make sure "Microsoft Edge WebView2 Runtime" is there.

4. **Restart Your Device:**
   - After you’ve checked, go ahead and restart your computer.

5. **Try Installing SEB Again:**
   - Once your device is back up, try running the SEB installer again.

Hopefully, this helps anyone else having the same issue! Feel free to share if you have other tips :)

**Reply by dbuechel** [reply]
The latest release version 3.8.0 does not contain the WebView2 dependency anymore, thus and as is our general recommendation, please try to use the latest version.
